flair/embeddings/transformer.py

```diff
@@ -372,32 +419,52 @@ def prepare_tensors(self, sentences: List[Sentence], device: torch.device = None
             device = flair.device
         flair_tokens, offsets, lengths = self.__gather_flair_tokens(sentences)
 
-        # encode inputs
-        batch_encoding = self.tokenizer(
-            flair_tokens,
-            stride=self.stride,
-            return_overflowing_tokens=self.allow_long_sentences,
-            truncation=self.truncate,
-            padding=PaddingStrategy.MAX_LENGTH if self.force_max_length else PaddingStrategy.LONGEST,
-            return_tensors="pt",
-            is_split_into_words=True,
-        )
-
-        tensor_args = self.__build_transformer_model_inputs(
-            batch_encoding, sentences, offsets, lengths, flair_tokens, device
-        )
-
-        return tensor_args
+        # random check some tokens to save performance.
```

Comment:
Why not check entire dataset or discard inputs not having all required metadata?

Reply:
As noted in the comment, running the checks on a broader scale would impact the speed, this is rather to ensure that the user gets a good warning if there are no bounding boxes in general, without impacting the performance.

And Discarding inputs would silently hide errors, so I am against that.